### ISSUE TITLE
[v3] Implement new Concept Exercise: classes

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#849]
This issue describes how to implement the `classes` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Classes in F#.

## Learning objectives

- Know what classes are.
- Know what the difference is between classes and records and when to use which.
- Know what encapsulation is.
- Know what fields are.
- Know how to create an object.
- Know how to define methods.

## Out of scope

- Static bindings.
- Mutable values.
- Properties.
- Composition.
- Additional constructors.
- `do` bindings in classes.
- `let` and `val` bindings in a class.
- Interfaces.
- Inheritance.
- Properties.
- Indexers.
- Structs.
- Method overloading.

## Concepts

The Concepts this exercise unlocks are:

- `classes`: know what classes are; know what the difference is between classes and records and when to use which; know what encapsulation is; know what fields are; know how to create an bject; 
know how to define methods; know how to update state.

## Prerequisites

This exercise's prerequisites Concepts are:

- `records`: know what records are.
- `tuples`: know what tuples are (used for parameter passing).
- `access-control`: know about the `public` and `private` access control 
- `unit`: know about the `unit` type.

Any data types used in this exercise (e.g. `strings` or `integers`) should also be added as prerequisites.

### Hints

- [Classes][classes]: introduction to classes.
- [Constructors][constructors]: how to define primary and additional constructors, and how to create objects by calling a constructor.
- [Methods][methods]: how to define methods in classes.
- [`let` bindings in classes][let-bindings-in-classes]: how to use `let` bindings to define fields.
- [`do` bindings in classes][do-bindings-in-classes]: how to use `do` bindings to add constructor logic.
- [`val` keyword][val-keyword]: describes `val` fields.
- [`unit` ][unit-type]: describes the `unit` type.

### After

- [Classes][classes]: introduction to classes.
- [Constructors][constructors]: how to define primary and additional constructors, and how to create objects by calling a constructor.
- [Methods][methods]: how to define methods in classes.
- [`let` bindings in classes][let-bindings-in-classes]: how to use `let` bindings to define fields.
- [`do` bindings in classes][do-bindings-in-classes]: how to use `do` bindings to add constructor logic.
- [`val` keyword][val-keyword]: describes `val` fields.
- [`unit` ][unit-type]: describes the `unit` type.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/classes.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[classes]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/classes
[constructors]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/constructors
[let-bindings-in-classes]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/let-bindings-in-classes
[do-bindings-in-classes]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/do-bindings-in-classes
[methods]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/methods
[val-keyword]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/members/explicit-fields-the-val-keyword
[unit-type]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/unit-type
